MyApplication/app/src/main/res/layout/fragment_main.xml

```diff
@@ -1,26 +1,39 @@
 <LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
     android:layout_marginBottom="?android:attr/actionBarSize"
-    android:gravity="center"
     android:orientation="vertical">
 
+    <include layout="@layout/gas_kilo_details" />
 
     <LinearLayout
+    android:id="@+id/resultsLL"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:visibility="invisible"
+    android:orientation="vertical">
+    <LinearLayout
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
+        android:gravity="bottom"
         android:layout_margin="10dp">
```

Comment:
All dp numbers should be inside dimens. 
Also, 
Write styles for each component you reuse....